### PR TITLE
Djanicek/auth debug

### PR DESCRIPTION
**Pull request by djanicekpach**
This updates k8s to the latest minikube-supported version: 1.27.4. https://github.com/kubernetes/minikube/releases

It turns out some in CI requests were sometimes being routed to the wrong pods/namespaces. The current theory is this was a bug in k8s since we were using an old version, and the upgrade helps. In testing, in 10 runs of the deploy tests we've had 1 failure, in it was unrelated to the dex or disconnect errors that have been plaguing us.

This requires some updates since PodSecurityPolicy is now deprecated and a couple Kubernetes resources moved from v1beta to v1.

### CODE REVIEW

File: etc/kubernetes-prometheus/manifests-all.yaml
Comment (jrockway) [inline etc/kubernetes-prometheus/manifests-all.yaml:1]: I wonder what happens if we delete this.  It's SUCH an old version of prometheus that is installed.  (There was a test that used it at one point, but I'm not sure if it's still around.)

File: etc/testing/circle/kube_debug.sh [hunk 1]
```diff
@@ -42,6 +43,29 @@ cmds=(
   'kubectl logs --namespace=test-cluster-6 --tail=1500 -l suite=pachyderm --all-containers=true --prefix=true'
   # if pachd restarted
   'kubectl logs --tail=1500 -l suite=pachyderm,app=pachd --previous --prefix=true'
+  'kubectl logs --namespace=test-cluster-1 --tail=1500 -l suite=pachyderm,app=pachd --previous --prefix=true'
```
Comment (jrockway) [inline etc/testing/circle/kube_debug.sh:46]: Any reason not to just grab suite=pachyderm?  Also, are there still 6 namespaces?  I feel like we might have dropped it down to 3 at some point.

File: etc/testing/gatekeeper.yaml [hunk 2]
```diff
@@ -2117,39 +2118,6 @@ metadata:
   name: gatekeeper-admin
   namespace: gatekeeper-system
 ---
-apiVersion: policy/v1beta1
```
Comment (jrockway) [inline etc/testing/gatekeeper.yaml:2120]: I think we might need to keep this (allows us to test against security configurations of customers), but I think the API just changed to policy/v1 maybe?

**Review by jrockway — APPROVED** [review]
Very nice.

File: etc/testing/gatekeeper.yaml (hunk 2 above)
Comment (djanicekpach) [reply etc/testing/gatekeeper.yaml:2120]: I think we are keeping the new equivalent way of doing it, but PSP was deprecated. https://kubernetes.io/docs/concepts/security/pod-security-policy/ 

The `pod-security.kubernetes.io/enforce: restricted` namespace annotation along with tweaks like `seccompProfile` are the replacement. I tried to follow the [migration plan](https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/) and set it to the most restrictive policy.

File: etc/testing/circle/kube_debug.sh (hunk 1 above)
Comment (djanicekpach) [reply etc/testing/circle/kube_debug.sh:46]: I'll switch to suite=pachyderm. I'm going to leave at 6 namespaces though. If I'm remembering right, I think there is a bug in minikubetestenv where the full-auto deployment and more manual InstallRelease count separately and we can end up with double the expected namespaces. I thought I wrote a ticket, but can't find it. I'll add one to the backlog if I can find it on a more thorough search.

File: .circleci/main.yml [hunk 3]
```diff
@@ -2250,7 +2250,7 @@ commands:
     steps:
       - run:
           name: Start minikube
-          command: minikube start --vm-driver=docker --kubernetes-version=v1.19.0 --cpus=7 --memory=12g --wait=all
+          command: minikube start --vm-driver=docker --kubernetes-version=v1.27.4 --cpus=7 --memory=12g --wait=all
```
[the remark this answers is not in the document]
Comment (djanicekpach) [reply .circleci/main.yml:2253]: Will make a ticket. This would be a great starter-story for a new build & release person.

**Pull request merged**